lib/DXIL/DxilModule.cpp

```diff
@@ -316,16 +316,13 @@ void DxilModule::CollectShaderFlagsForModule(ShaderFlags &Flags) {
 
   const ShaderModel *SM = GetShaderModel();
 
-  unsigned NumUAVs = m_UAVs.size();
+  unsigned NumUAVs = 0;
   const unsigned kSmallUAVCount = 8;
-  if (NumUAVs > kSmallUAVCount)
-    Flags.Set64UAVs(true);
-  if (NumUAVs && !(SM->IsCS() || SM->IsPS()))
-    Flags.SetUAVsAtEveryStage(true);
 
   bool hasRawAndStructuredBuffer = false;
 
   for (auto &UAV : m_UAVs) {
+    NumUAVs += UAV->GetRangeSize();
```

Comment:
Need to watch for overflow.  Unbounded will be UINT_MAX for GetRangeSize() and should automatically be larger than kSmallUAVCount, yet it would overflow and reduce the count by one here.